### ISSUE TITLE
MNIST target_transform

### ISSUE THREAD

**Issue by achainan** [pytorch/vision#423]
In the MNIST dataset the target is a label (not a PIL image). Hence trying to do a `target_transform` wouldn't really work right?

https://github.com/pytorch/vision/blob/13d1a6cc955ca9a49f240a99b44ddcc8a685efbc/torchvision/datasets/mnist.py#L67-L82

Shouldn't we get rid of the `target_transform` input?
Alternatively we could do the following if there is a `target_transform`:

```
    def __getitem__(self, index):
        """
        Args:
            index (int): Index
        Returns:
            tuple: (image, target) where target is index of the target class.
        """
        if self.train:
            img, target = self.train_data[index], self.train_labels[index]
        else:
            img, target = self.test_data[index], self.test_labels[index]

        # doing this so that it is consistent with all other datasets
        # to return a PIL Image
        image = Image.fromarray(img.numpy(), mode='L')

        if self.transform is not None:
            img = self.transform(image)

        if self.target_transform is not None:
            target = self.target_transform(image)

        return img, target
```

**Comment by davidlmorton** [comment]
My understanding is that transforms are just callables and not restricted in their input/output types.  So `target_transform` on the MNIST dataset could be something that takes in an integer and outputs a onehot tensor or something like that.  I think it is nice symmetry, and they just haven't included any transforms (in torchvision.transforms) that aren't image related.
